src/tensorflow/lite/micro/tools/project_generation/create_tflm_arduino.py

```diff
@@ -0,0 +1,548 @@
+# Lint as: python3
+# Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Script to generate TFLM Arduino examples ZIP file"""
+
+import argparse
+import logging
+import shutil
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import Dict, Iterable, List, Set, Tuple, Union
+from enum import Enum, unique
+import configparser
+
+# parse args: {outdir}
+# generate list of all source/header files and their src/dst locations
+# create directory tree in {outdir}/tflm_arduino
+# download third-party source
+# copy templates: library.properties to {outdir}/tflm_arduino
+# copy templates: TensorFlowLite.h to {outdir}/tflm_arduino/src
+# copy transforms: examples
+#   --platform=arduino
+#   --third_party_headers=<output from make
+#       list_{example_name}_example_headers
+#       list_third_party_headers>
+#   --is_example_source
+# copy transforms: examples/{example_name}/main_functions.cc to
+#  {example_name}.ino
+#   --platform=arduino
+#   --third_party_headers=<output from make
+#       list_{example_name}_example_headers
+#       list_third_party_headers>
+#   --is_example_ino
+# copy transforms: tensorflow, third_party
+#   --platform=arduino
+#   --third_party_headers=<output from make
+#       list_third_party_headers>
+# patch third_party/flatbuffers/include/flatbuffers/base.h with:
+#   sed -E 's/utility\.h/utility/g'
+# patch third_party/kissfft/kiss_fft.h
+#   sed -E 's@#include <string.h>@//#include <string.h>
+# 	 /* Patched by helper_functions.inc for Arduino compatibility */@g'
+# run fix_arduino_subfolders.py {outdir}/tflm_arduino
+# remove all empty directories in {outdir}/tflm_arduino tree
+# create ZIP file using shutil.make_archive()
+
+
+def RunSedScripts(file_path: Path,
+                  scripts: List[str],
+                  args: Union[str, None] = None,
+                  is_dry_run: bool = True) -> None:
+  """
+  Run SED scripts with specified arguments against the given file.
+  The file is updated in place.
+
+  Args:
+    file_path: The full path to the input file
+    scripts: A list of strings, each containing a single SED script
+    args: a string containing all the SED arguments | None
+    is_dry_run: if True, do not execute any commands
+  """
+  cmd = "sed"
+  if args is not None and args != "":
+    cmd += f" {args}"
```

Comment:
combining string concatenation with f-strings seems to defeat the purpose of f-strings.